src/components/multi-select/multi-select.test.jsx

```diff
@@ -0,0 +1,516 @@
+  findSelectedOption,
+  openOptions,
+  queryAvailableOption,
+  querySelectedOption,
+  queryRemoveButton,
+} from './test-queries.js';
+
+describe('<MultiSelect>', () => {
+  const requiredProps = {
+    options: ['Foo', 'Bar'],
+    id: 'test-id',
+    label: 'test label',
+    name: 'field-id',
+  };
+
+  it('has defaults', () => {
+    const ref = createRef();
+
+    render(<MultiSelect {...requiredProps} ref={ref} />);
+    expect(document.body).toMatchSnapshot();
+    expect(ref.current).toMatchSnapshot();
+  });
+
+  describe('classNames API', () => {
+    it('uses class name overrides provided', () => {
+      render(<MultiSelect
+        {...requiredProps}
+        classNames={{
+          container: 'unique-container',
+          formControlChildrenContainer: 'unique-form-control-children-container',
+          input: 'unique-input-class',
+          tagList: 'unique-tag-list-class',
+        }}
+      />);
+
+      expect(document.querySelector('.unique-container')).toBeInTheDocument();
+      expect(document.querySelector('.unique-form-control-children-container')).toBeInTheDocument();
+      expect(document.querySelector('.unique-input-class')).toBeInTheDocument();
+      expect(document.querySelector('.unique-tag-list-class')).toBeInTheDocument();
+    });
+  });
+
+  describe('id API', () => {
+    it('sets id of container and selected options', async () => {
+      render(<MultiSelect {...requiredProps} id="unique-id" value={['Foo']} />);
+
+      expect(document.querySelector('#unique-id')).toBeInTheDocument();
+      expect((await findSelectedOption('test label', 'Foo')).parentElement).toHaveAttribute('id', 'unique-id-option-Foo');
+    });
+  });
+
+  describe('label API', () => {
+    it('sets label of form control and clear button', async () => {
+      render(<MultiSelect {...requiredProps} label="unique label" value={['Foo']} />);
+
+      expect(screen.getByText('unique label')).toBeInTheDocument();
+      expect(await findRemoveButton('unique label', 'Foo')).toBeInTheDocument();
+    });
+  });
+
+  describe('listboxChildren API', () => {
+    it('uses render function for listbox children if set', async () => {
+      function listboxChildren(options, refs, onSelect) {
+        return (
+          <div>
+            {options.map((option, index) => (
+              <span key={option} ref={refs[index]} onClick={() => { onSelect({ target: refs[index] }); }}>Override {option}</span>
+            ))}
+          </div>
+        );
+      }
+
+      render(<MultiSelect {...requiredProps} listboxChildren={listboxChildren} />);
+
+      await openOptions('test label');
+      expect(screen.getByText('Override Foo')).toBeInTheDocument();
+      userEvent.click(screen.getByText('Override Foo'));
+      expect(await findSelectedOption('test label', 'Foo')).toBeInTheDocument();
+    });
+  });
+
+  describe('onChange API', () => {
+    it('fires the onChange event when the value changes', async () => {
+      let value = [];
```

Comment:
This is odd. The standard patterns for testing callbacks is to use spies.

e.g. 

```
const spy = jest.fn();
render(<Component onInput={spy} />)
userEvent...
expect(spy).toHaveBeenCalledWith({ target: { value: ... }});
```

Reply:
See this comment: https://github.com/mavenlink/design-system/pull/311#discussion_r640196456